maat-court-data-api/src/main/java/gov/uk/courtdata/repository/StoredProcedureRepository.java

```diff
@@ -170,15 +170,34 @@ private boolean connectionRequiresManualCommit(Connection connection) throws SQL
         return !connection.getAutoCommit();
     }
 
-    private boolean rollbackTransactionalConnection(Connection connection) throws SQLException {
+    private void rollbackTransactionalConnection(Connection connection) throws SQLException {
         try {
             if (connectionRequiresManualCommit(connection)) {
                 connection.rollback();
             }
-            return true;
         } catch (SQLException se) {
             log.error(getClass().getName() + " ### Cannot rollback transaction ");
             throw se;
         }
     }
+
+    public void handleException(Throwable t, Connection conn) throws SQLException, MAATApplicationException {
+        log.error("### " + t);
+
+        rollbackTransactionalConnection(conn);
+        if (t instanceof SQLException sqlException) {
+            int errorCode = sqlException.getErrorCode();
+            if ((errorCode >= MIN_SQL_ERROR_CODE) && (errorCode <= MAX_SQL_ERROR_CODE)) {
+                String errorMessage = sqlException.getMessage();
+                //remove stack trace codes from exception
+                int start = errorMessage.indexOf(':');
+                int end = errorMessage.indexOf('\n');
+                if (start > 0 && end > 0) {
+                    errorMessage = "Application Error " + errorMessage.substring(start, end);
+                }
+                throw new MAATApplicationException(errorMessage, sqlException);
+            }
+        }
+    }
```

Comment:
Just a curious question, why the need to only handle the sql errors in the min - max range? If I understand right anything out of this range will still be handled by the default handler and returned as 500 with the exception message, just these have the message tidied up a bit?

Reply:
Good question, so this specific range of  SQL error codes are being used to raise application errors in the stored procedure, such as validation errors, that need to be displayed on the UI. The application errors are identified with new error code, all other DB errors the generic error message will be displayed.